### PR TITLE
feat(android-sdk): fusionauth android sdk beta blog and site update

### PR DESCRIPTION
**Pull request by Aaron-Ritter**
This is the first draft for the FusionAuth Android SDK Beta Release.

The fusionauth-site edits and blogpost are from the perspective of releasing a FusionAuth Android SDK Beta, and the status is currently as follow:
- https://github.com/FusionAuth/fusionauth-quickstart-java-android-fusionauth-sdk needs to be renamed to fusionauth-quickstart-kotlin-android-fusionauth-sdk as its written in kotlin

### CODE REVIEW

**Comment by alex-fusionauth** [comment]
@Aaron-Ritter the file layout should match our other quickstart like this one
https://github.com/FusionAuth/fusionauth-quickstart-java-android-native

Having the docker image and kickstart folder in the root and then the `complete-application` contain the android app.

It might also be simpler to create a new repo with the correct naming.

**Comment by Aaron-Ritter** [comment]
> @Aaron-Ritter the file layout should match our other quickstart like this one https://github.com/FusionAuth/fusionauth-quickstart-java-android-native
> 
> Having the docker image and kickstart folder in the root and then the `complete-application` contain the android app.

Sure, we can easily change that by moving most of it in to the complete-application sub-directory. From a handling perspective inside Android Studio it was simpler to have the project in the root directory but it should work with the other structure too.

> It might also be simpler to create a new repo with the correct naming.

I currently have write access to this https://github.com/FusionAuth/fusionauth-quickstart-java-android-fusionauth-sdk and the android-fusionauth-sdk. so if we could just rename fusionauth-quickstart-java-android-fusionauth-sdk to fusionauth-quickstart-kotlin-android-fusionauth-sdk the access can stay for now as it is without a change required.

**Comment by alex-fusionauth** [comment]
> I currently have write access to this https://github.com/FusionAuth/fusionauth-quickstart-java-android-fusionauth-sdk and the android-fusionauth-sdk. so if we could just rename fusionauth-quickstart-java-android-fusionauth-sdk to fusionauth-quickstart-kotlin-android-fusionauth-sdk the access can stay for now as it is without a change required.

@Aaron-Ritter 

Here is the full docs for quickstarts
https://github.com/FusionAuth/fusionauth-example-template/blob/master/QUICKSTART-INSTRUCTIONS.md

For naming this should follow the same, I have put in a request to Change name from `fusionauth-quickstart-java-android-fusionauth-sdk` to `fusionauth-quickstart-kotlin-android-native`.

**Comment by Aaron-Ritter** [comment]
@alex-fusionauth I've already moved the contents of the quickstart and will update the documentation accordingly and request a new review accordingly. We originally based it on it but were modifying it for a potential easier automated release later on but we will cross that bridge when we come to it.

Does it make sense what i meant by renaming the quickstart repository? 

Are you still reviewimg the rest of the fusionauth-site PR (Blog/SDK) or do you have additional input i can consider while reworking the stuff related to quickstart?

**Comment by Aaron-Ritter** [comment]
@alex-fusionauth we are currently working on the FusionAuth  1.50.1 upgrade, we fixed most of it but are still looking into some unexpected behaviour https://github.com/FusionAuth/fusionauth-android-sdk/pull/60. 

So if you don't find anything else which needs changing we'd suggest to still wait for the merge of the PR until we fixed the incompatibility with the latest FusionAuth release and created a new version of the SDK.

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (alex-fusionauth) [inline astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:52]: The `applicationId` is incorrect as it doesn't follow the others quickstart.

Should be 
`e9fdb985-9173-4e01-9d73-ac2d60d1dc8e`

Docker should also use 

```
  fusionauth:
    image: fusionauth/fusionauth-app:latest
```

**Review by alex-fusionauth — CHANGES_REQUESTED** [review]
Small issues, some due to renaming the repo and others need to follow the guide for quickstart to get correct docker instructions.

I ran with 1.49.2 and get this after following the instructions
image

File: astro/src/content/blog/android-sdk-beta.mdx
Comment (Aaron-Ritter) [reply astro/src/content/blog/android-sdk-beta.mdx:38]: the suggested change is the same, i guess you want to change best practice defaults as this might be a bit confusing?

File: astro/src/content/docs/sdks/android-sdk.mdx
Comment (Aaron-Ritter) [reply astro/src/content/docs/sdks/android-sdk.mdx:17]: This is something we tested thoroughly, it's how to work locally with android emulator. And it's how the automated workflow in github works.

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:59]: Not sure what's the suggested change.

File: astro/src/content/quickstarts/quickstart-android-java-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-java-native.mdx:2]: Because both the AppAuth quickstart and FusionAuth SDK quickstart is live at the same time it just makes sense to specify it more clearly especially as its used in the overview of quickstarts.

File: astro/src/content/blog/android-sdk-beta.mdx
Comment (Aaron-Ritter) [reply astro/src/content/blog/android-sdk-beta.mdx:2]: we need as well a blog image, we don't create this :)

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:52]: the kickstart is not a usual one in any case as it covers more use cases.

The image i would prefer to leave on the latest tested version.

We have a manual release process of the quickstart originating from the SDK which has a fixed version, which we have planned to automate once all other workflows are stable.

File: astro/src/content/blog/android-sdk-beta.mdx
Comment (Aaron-Ritter) [reply astro/src/content/blog/android-sdk-beta.mdx:85]: updated

File: astro/src/content/blog/android-sdk-beta.mdx
Comment (Aaron-Ritter) [reply astro/src/content/blog/android-sdk-beta.mdx]: sorry, IDE was not showing the difference obviously. merged.

File: astro/src/content/docs/sdks/android-sdk.mdx
Comment (Aaron-Ritter) [reply astro/src/content/docs/sdks/android-sdk.mdx:32]: updated

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx]: sorry, IDE was not showing the difference obviously. merged.

File: astro/src/content/docs/sdks/android-sdk.mdx
Comment (Aaron-Ritter) [reply astro/src/content/docs/sdks/android-sdk.mdx:17]: We have the TESTING doc we're working on which deep dives in to E2E testing which makes use of the 10.0.2.2 IP and stuff like ngrok would then again cause issues with it. So to avoid confusion we will further explain what that IP means (its basically the host address from the perspective of the Android Emulator) and in what context this IP is important/used.

**Comment by Aaron-Ritter** [comment]
@alex-fusionauth added a few comments, otherwise mostly merged and updated a few other additional wrong links/references. 

The `Not authorized` issue mentioned by you above is something which we did not experience during our testing (if you checkout the repository and start it in Android Studio it should just work). 

Once the new SDK version is released which is at the moment pending on https://github.com/FusionAuth/fusionauth-android-sdk/issues/64, i will fully re-test the quickstart with 1.50.1.

**Comment by Aaron-Ritter** [comment]
We're good to go. @alex-fusionauth

**Review by mark-robustelli — CHANGES_REQUESTED** [review]
Overall, I got it running. I could figure out what need to be created where and got it going. It could be because I am not a seasoned Android developer, but it was not always clear to me why I was doing what I was doing.  The copy and paste is fine, but I do think there needs to be some explanation. 

Also, I see for the blog that you mentioned that you do not create the art work. Do you know if Alex started that process? I can if not.

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:83]: in a real example this would be likely even a more complex configuration with a variable which builds an app against the dev, test and prod fusionauth server. Not sure if we need to explain this in more detail in the quickstart.

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:52]: this was a merge problem, i fixed it in all previous versions but not the latest, its all fixed now.

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:54]: added a note to just open complete-application in android studio

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:65]: linked to the official instructions [FusionAuth SDK as a dependency](https://central.sonatype.com/artifact/io.fusionauth/fusionauth-android-sdk/)

File: astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx
Comment (Aaron-Ritter) [reply astro/src/content/quickstarts/quickstart-android-kotlin-native.mdx:89]: Not adding. It's a Beta release I'd expect experienced developers to go through this, and in addition you'd be able to use any IDE as long its supports the used versions which we don't explain either.

**Comment by Aaron-Ritter** [comment]
Thanks a lot for the additional review @mark-robustelli I've updated most of your feedback, and as well fixed the styling which was fixed for all previous versions but not 1.51 css. I've updated https://github.com/FusionAuth/fusionauth-android-sdk/?tab=readme-ov-file#release-process making sure the quickstart files are updated with each release.

> Overall, I got it running. I could figure out what need to be created where and got it going. It could be because I am not a seasoned Android developer, but it was not always clear to me why I was doing what I was doing. The copy and paste is fine, but I do think there needs to be some explanation.

All thought we based the quickstart on the existing AppAuth one we could go in to more detail. But i strongly suggest to do that in a later step once we have a first v1.0.0 SDK release and rather consider improving the documentation around the SDK first and then reference the documentation from the quickstart.

> Also, I see for the blog that you mentioned that you do not create the art work. Do you know if Alex started that process? I can if not.

I dont think he started anything, yes please create it :+1: .

**Review by alex-fusionauth — APPROVED** [review]
Adjusted what was needed, this is good to go!

**Comment by alex-fusionauth** [comment]
Closed by #3123

**Pull request closed without merging**
